tests/conftest.py

```diff
@@ -327,3 +333,56 @@ def render_span(span):
         return "\n".join(render_span(root_span))
 
     return inner
+
+
+@pytest.fixture(name="Any")
+def make_any():
+    """
+    An object which matches any instance of the class passed to the object at
+    instantiation time.
+
+    Useful for assert_called_with statements.
+    """
+
+    class Any(object):
+        def __init__(self, target_class):
+            self.target_class = target_class
+
+        def __eq__(self, other):
+            return isinstance(other, self.target_class)
+
+    return Any
+
+
+@pytest.fixture(name="StringContaining")
+def make_string_containing():
+    """
+    An object which matches any string containing the substring passed to the
+    object at instantiation time.
+
+    Useful for assert_called_with statements.
+    """
+
+    class StringContaining(object):
+        def __init__(self, substring):
+            self.substring = substring
+
+        def __eq__(self, test_string):
+            return self.substring in test_string
+
+    return StringContaining
+
+
+@pytest.fixture(name="FunctionMock")
+def function_mock():
+    """
+    Just like a mock.Mock object, but one which always passes an isfunction
+    test.
+    """
+
+    class FunctionMock(mock.Mock):
+        def __init__(self, *args, **kwargs):
+            super(FunctionMock, self).__init__(*args, **kwargs)
+            self.__class__ = FunctionType
+
+    return FunctionMock
```

Comment:
> I don't know how you're going to use mock but I've just been creating my own stub classes. I find a lot of mock's behavior too magical wrt attribute access on the mock obj

Here, for instance:

```python
def test_passes_custom_samling_context_from_start_transaction_to_traces_sampler(
    FunctionMock, sentry_init, DictionaryContaining
):
    traces_sampler = FunctionMock()
    sentry_init(traces_sampler=traces_sampler)

    start_transaction(custom_sampling_context={"dogs": "yes", "cats": "maybe"})

    traces_sampler.assert_any_call(
        DictionaryContaining({"dogs": "yes", "cats": "maybe"})
    )
```

Would rather not have to rewrite the machinery of tracking calls.

Reply:
```python
def test_passes_custom_samling_context_from_start_transaction_to_traces_sampler(
    sentry_init
):
    sampler_calls = []
    sentry_init(traces_sampler=sampler_calls.append)

    start_transaction(custom_sampling_context={"dogs": "yes", "cats": "maybe"})

    assert sampler_calls == [{"dogs": "yes", "cats": "maybe"}]
```

Unless you need to account for additional keys or calls I would go with this...

Reply:
Feel free to add... I am not entirely sure why people keep using this stuff but I don't want to fight it.

Reply:
Okay, in the interest of time I'm going to leave it for now, but that's an interesting pattern to know about, thanks!